backend/src/server.js

```diff
@@ -1,87 +1,315 @@
+    return passport.authenticate('jwt', { session: 'false' }, async function (err, user) {
+        if (err)
+            next(err);
+        else if (user === false) {
+            let resBody = errors.makeAuthorizationError([{ id: "invalidToken", properties: [] }]);
+            res.status(401).send(resBody);
+        }
+        else if (user === null) {
+            let resBody = errors.makeAuthorizationError([{ id: "notFound", properties: ["_id"] }]);
+            res.status(401).send(resBody);
+        }
+        else {
+            let char = await klad.getByPropFromKladovka(COLLECTIONS.CHARS, 'user_id', user._id);
+            if (char === null)
+                res.sendStatus(204);
+            else
+                res.status(200).send(char);
+        }
+    })(req, res, next);
+});
+
+app.delete(API.CHARS, async function (req, res, next) {
+    return passport.authenticate('jwt', { session: 'false' }, async function (err, user) {
+        if (err)
+            next(err);
+        else if (user === false) {
+            let resBody = errors.makeAuthorizationError([{ id: "invalidToken", properties: [] }]);
+            res.status(401).send(resBody);
+        }
+        else if (user === null) {
+            let resBody = errors.makeAuthorizationError([{ id: "notFound", properties: ["_id"] }]);
+            res.status(401).send(resBody);
+        }
+        else {
+            let deleted_count = await klad.deleteAllByPropFromKladovka(COLLECTIONS.CHARS, 'user_id', user._id);
+            res.status(200).send({ deleted_count });
+        }
+    })(req, res, next);
+});
+
+app.post(API.USERS, async function (req, res) {
+    let user = req.body;
+    let validationResult = await validation.checkSignUp(user);
     if (validationResult.isValid) {
-        let added_id = await klad.placeInKladovka(db, Collections.Chars, validationResult.char);
-        res.header('Location', urlJoin(API.CHARS, added_id));
-        res.status(201).send({ added_id });
+        let verificationResult = await verification.checkEmail(user.email);
+        if (verificationResult.isVerify) {
+            let readyUser = User.readyToSave(validationResult.user);
+            let added_id = await klad.addInKladovka(COLLECTIONS.USERS, readyUser);
+            res.header('Location', urlJoin(API.USERS, added_id));
+            res.status(201).send({ added_id });
+        } else {
+            let resBody = errors.makeVerificationError(verificationResult.errors);
+            res.status(400).send(resBody);
+        }
     } else {
         let resBody = errors.makeValidationError(validationResult.errors);
         res.status(400).send(resBody);
     }
 });
 
-app.put(urlJoin(API.CHARS, ':id'), async function (req, res) {
-    let char = req.body;
-    let validationResult = validation.checkChar(char);
+app.post(API.TOKENS, async function (req, res, next) {
+    let user = req.body;
+    let validationResult = await validation.checkSignIn(user);
```

Comment:
I was talking about the verify callback of 'local' strategy in passport.js. You should put validation for signin there. If it fails, you should create an `err` and then `done(err)`. Then it will be caught by the final error handler and proper response will be composed.